src/osip-simulation-view/src/main/java/edu/kit/pse/osip/simulation/view/dialogs/OverflowDialog.java

```diff
@@ -1,16 +1,85 @@
 
 import edu.kit.pse.osip.core.model.base.TankSelector;
+import edu.kit.pse.osip.core.utils.language.Translator;
+import javafx.geometry.Insets;
+import javafx.scene.Scene;
+import javafx.scene.image.Image;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Priority;
+import javafx.scene.text.Font;
+import javafx.scene.text.Text;
+import javafx.scene.text.TextFlow;
+import javafx.stage.Stage;
+import javafx.scene.paint.Color;
+import edu.kit.pse.osip.simulation.view.main.ViewConstants;
+import javafx.geometry.VPos;
 
 /**
  * This window informs the user, that a tank has an overflow.
+ * 
+ * @author Maximilian Schwarzmann
+ * @version 1.0
  */
-public class OverflowDialog {
+public class OverflowDialog extends Stage {
+    private TankSelector tank = null;
+    private static final int FONT_SIZE = ViewConstants.FONT_SIZE;
+    private static final double MIN_WINDOW_HEIGHT = 200;
+    private static final double MIN_WINDOW_WIDTH = 500;
+    
+    /**
+     * Constructor of OverflowDialog
+     */
+    public OverflowDialog() {
+        configure();
+    }
+    
+    
     /**
      * Set the tank which has an overflow
      * @param tank The tank which has an overflow
      */
-    public final void setTank (TankSelector tank) {
-        throw new RuntimeException("Not implemented!");
+    public final void setTank(TankSelector tank) {
+        this.tank = tank;
+        configure();
     }
+
+    private Text getTextWithTankName(String id, TankSelector tank, Font font) {
+        Text text = new Text();
+        if (tank != null) {
+            text.setText(Translator.getInstance().getString(id) + " " + tank.toString() + "!" + "\n");
+        } else {
+            throw new IllegalStateException("No tank set. Please call setTank() before calling show()");  
+        }               
+        if (font != null) {
+            text.setFont(font);
+        }
+        return text;
+    }
+    
+    private void configure() {
+        this.setTitle(Translator.getInstance().getString("simulation.overflowdialog.title"));
+        Image imageIcon = new Image(getClass().getClassLoader().getResourceAsStream("icon.png"));
+        this.getIcons().add(imageIcon);
+        GridPane grid = new GridPane();
+        grid.setPadding(new Insets(ViewConstants.ELEMENTS_GAP));
+        
+        TextFlow overflowTextFlow = new TextFlow();
+        Text overflowText = getTextWithTankName("simulation.overflowdialog.overflowmessage", tank,
+                new Font(FONT_SIZE * 3));
+        overflowText.setFill(Color.RED);
+        overflowTextFlow.getChildren().add(overflowText);
+        GridPane.setConstraints(overflowTextFlow, 0, 0);
+        GridPane.setValignment(overflowTextFlow, VPos.CENTER);
+        
+        grid.getChildren().addAll(overflowTextFlow);
+        Scene scene = new Scene(grid, MIN_WINDOW_WIDTH, MIN_WINDOW_HEIGHT);
+        
+        grid.setHgrow(overflowTextFlow, Priority.ALWAYS);
```

Comment:
Das hier ist auch ein statischer Aufruf. Daher empfehle ich hier, diesen auch auf GridPane direkt zu machen. Ich denke, es bietet sich auch an, den Aufruf unter die Aufrufe in Zeile 72 und 73 zu stellen.